### DISCUSSION TITLE
Curious about different advertising behaviors

### DISCUSSION THREAD

**Discussion by xerothermic** [pybricks/support#825] [Q&A] [answered]
Hi,

I got a chance to try PyBricks firmware on my RI Hub yesterday and everything worked smoothly. I have a M1 MacBook and downloaded Chrome to follow along the Wireless communication tutorial to drive the robot using Bleak. 

I noticed after I flashed the PyBricks firmware, I could immediately discover the device name "PyBricks Hub" name and connect to it.

Then, after reading lwp3 doc, I thought to repeat the wireless communication experiment with the official RI firmware. However, after switching back to official firmware. I don't see any BleakScanner results that is similar to "PyBricks dev" that I can connect to. 

I saw there is a `pybricksdev lwp3 repl` cli to connect to official firmware. Unfortunately, it's currently not supported due to lack of m1 support. I looked at the python file, but it seems like the connection logic is also using BleakScanner. The program filters advertising service UUID to match LEGO one. I printed service UUID for all scanned devices, but they were all empty.

The strange thing is that MINDSTORMS app could still discover and connect to the official firmware. 

I suspect the official firmware doesn't send advertising packets as the PyBricks firmware do?

Any suggestion on how to connect to the official firmware via BleakScanner, BleakClient?

**Comment by laurensvalk** [comment, marked as answer]
Thanks for reaching out. Great to hear that installing Pybricks and using the wireless demos worked smoothly! 

The original firmware uses Bluetooth Classic instead of Bluetooth Low Energy as used in Pybricks and Bleak.

You can probably connect to the hub with the original firmware using your MacBook’s Bluetooth settings. Make sure to press the Bluetooth button on the hub to activate Bluetooth. Afterwards, you’ll then be able to exchange data with the hub through a virtual comm port.

**Reply by xerothermic** [reply]
Hi @laurensvalk,

Thanks for your explanation. I got confused when I saw #353 and mistakenly thought that it will also work for 51515 original firmware, too. Now, you explained the original firmware is only using Bluetooth Classic instead of BLE, then it makes sense that we won't discover the Hub with Bleak and by scanning for GATT adv packets...

I have a follow up question:
> you’ll then be able to exchange data with the hub through a virtual comm port.

Are you referring to the interactive REPL that's available when connected to official firmware via the virtual COM port?

Thanks again for your reply!

**Reply by laurensvalk** [reply]
That's right. You may need to hit CMD+C (or send binary 3) a few times to exit the runtime and drop into the REPL.

You'd have to do a bit of work to combine running a script with data exchange in order to get a result similar to the pybricks communication example.

There's a paste mode in the REPL to get a script across, for example. Type help() in the REPL to see how.
